### DISCUSSION TITLE
Custom Detection Models

### DISCUSSION THREAD

**Discussion by toadee** [blakeblackshear/frigate#6732] [Ask A Question] [answered]
Is it possible to setup ones own custom models for tensorrt?

I recently set mine up but frigate is detecting trucks as cars, my dogs as cats and, recently, a horse as a dog.

I've read it's possible to train your own models (seen here: https://blog.paperspace.com/yolov7/) but I wouldn't know where the models file is placed.

Is it possible the poor detections is due to a lack of models file? Got frigate running in docker on unRAID

**Comment by NickM-27** [comment]
of course you can run your own model, just the same as the models that you generate with tensorrt. 

> Is it possible the poor detections is due to a lack of models file? 

It's not clear what you mean here, there are model files otherwise the model couldn't be used...?

**Reply by toadee** [reply]
I was just not sure what the model file extensions would be. I've seen some readings where mention of a .txt file is made others referring to .pt file but I assume the trt files are then the models themselves.

**Reply by NickM-27** [reply, marked as answer]
The labelmap is .txt and the model depends on the detector being used. Could be .trt, .tflite, .xml, etc
